test/worker_test.py

```diff
@@ -2135,3 +2136,61 @@ def run(self):
                 self.assertEqual(0, self.sch._state.get_task(s1.task_id).failures.num_failures())
                 self.assertEqual(self.per_task_retry_count, self.sch._state.get_task(e2.task_id).failures.num_failures())
                 self.assertEqual(self.default_retry_count, self.sch._state.get_task(e1.task_id).failures.num_failures())
+
+
+class SleepyLocalTarget(luigi.LocalTarget):
```

Comment:
Can we reduce the complexity of the testcase you're building? I'm not sure I fully understand what you're solving and that maybe could be easier if the test case was simpler (if possible).